### DISCUSSION TITLE
OIDC Differences between  3.10 & 3.11 release ?

### DISCUSSION THREAD

**Discussion by Vinche59** [quarkusio/quarkus#41364] [Q&A] [answered]
Hello,

Is there any modification for OIDC management for rest-client in Quarkus 3.11 release (compared to 3.10 version) ?
I didn't notice anything specific in the release notes and migration guide but when I tried to update my project from Quarkus 3.10.2 to Quarkus 3.11.3, I've this failure:

`2024-06-21T16:35:38.077208834Z Caused by: io.smallrye.mutiny.CompositeException: Multiple exceptions caught:
2024-06-21T16:35:38.077211645Z 	[Exception 0] io.quarkus.runtime.configuration.ConfigurationException: 'quarkus.oidc.client.auth-server-url' property must be configured
[...]
Caused by: [CIRCULAR REFERENCE: io.quarkus.runtime.configuration.ConfigurationException: 'quarkus.oidc.client.auth-server-url' property must be configured]`

In application.properties, I've got `quarkus.oidc-client.auth-server-url` defined.
Should I change it to `quarkus.oidc.client.auth-server-url`  as stated in the error message ?

[a comment by an automated account is not shown]

**Comment by sberyozkin** [comment]
@Vinche59 I'm not sure why this error message is reporting `.` instead of `-`, see https://github.com/quarkusio/quarkus/blob/3.11.3/extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonUtils.java#L92

Looks like there is some mix up, but hard to advise without a reproducer, please create the one if you can
thanks

**Reply by sberyozkin** [reply]
Sorry, wrong property, I can only see https://github.com/quarkusio/quarkus/blob/3.11.3/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java#L346, and there is no similar error in https://github.com/quarkusio/quarkus/blob/3.11.3/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java

So I don't now what is going on, having a reproducer would help

**Reply by sberyozkin** [reply]
@Vinche59 I think I may have something similar reproduced. but related to `quarkus.oidc.client-id`... Can you please prepare a simple reproducer ?

**Reply by sberyozkin** [reply]
That was a false alarm re the client id...

**Comment by Vinche59** [comment, marked as answer]
Hello,

Sorry for the late reply, I was on PTO.
Actually, while trying to create a reproducer, I found my mistake...
The problem reported actually only happened when deploying on our K8S infra and seems to come from "bad escaping" of the env variables.

Before 3.11, this env variable was pickup correctly:
QUARKUS_OIDC_CLIENT_EASYTRACK_CLIENT_AUTH_SERVER_URL

After 3.11, I had to add a double underscore:
QUARKUS_OIDC_CLIENT__EASYTRACK_CLIENT__AUTH_SERVER_URL

**Reply by sberyozkin** [reply]
@Vinche59 Hi, np, right, I see it documented at https://quarkus.io/guides/config-reference#system-properties, which applies to a case like this one where you have a named OIDC client, good it works now for you

**Reply by Vinche59** [reply]
Yes, it's the doc that helps me while trying to understand why it was working on quarkus:dev and not in my K8S env :-)
Really sorry for the noise and disturbance...
